src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,110 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (key == null) {
+            putForNullKey(value);
+        } else {
+            size = addEntry(Objects.hashCode(key), key, value, getIndex(key)) ? ++size : size;
+        }
+        resize();
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> currentNode = table[getIndex(key)];
+        while (currentNode != null) {
+            if (Objects.equals(key, currentNode.key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(Object key) {
+        return (key == null) ? 0 : Objects.hashCode(key) & table.length - 1;
```

Comment:
Let's make it easier. 
```suggestion
        return key == null ? 0 : key.hashCode() % table.length ;
```

Reply:
UPD: Remember that there may be negative numbers here use Math.abs().

Reply:
+